ledger-tool/src/main.rs

```diff
@@ -943,6 +996,108 @@ fn main() {
                 }
             }
         }
+        ("verify-snapshot", Some(arg_matches)) => {
+            load_bank_from_snapshot(arg_matches, &ledger_path, true, true);
+        }
+        ("convert-accounts", Some(arg_matches)) => {
+            let output_directory = value_t_or_exit!(arg_matches, "output_directory", String);
+            let mut root_bank = load_bank_from_snapshot(arg_matches, &ledger_path, false, false);
+            // Convert root_bank.EpochStakes
+            for (_, stakes) in root_bank.epoch_stakes.iter_mut() {
+                for (pubkey, (_, vote_account)) in stakes.vote_accounts.iter_mut() {
+                    VoteStateVersions::convert_from_raw(vote_account, &pubkey);
+                }
+            }
+
+            for (pubkey, (_, vote_account)) in
+                root_bank.stakes.write().unwrap().vote_accounts.iter_mut()
+            {
+                VoteStateVersions::convert_from_raw(vote_account, &pubkey);
+            }
+
+            let index: Vec<_> = {
+                let index = root_bank
+                    .rc
+                    .accounts
+                    .accounts_db
+                    .accounts_index
+                    .read()
+                    .unwrap();
+
+                // Write out the new accounts
+                let total: usize = index
+                    .account_maps
+                    .values()
+                    .map(|slot_list| slot_list.read().unwrap().len())
+                    .sum();
+
+                println!("Converting {} accounts", total);
+                index
+                    .account_maps
+                    .iter()
+                    .map(|(pubkey, slot_list)| (*pubkey, slot_list.read().unwrap().clone()))
+                    .collect()
+            };
+
+            for (pubkey, slot_list) in index.iter() {
+                for (slot, _) in slot_list.iter() {
+                    let ancestors = vec![(*slot, 1)].into_iter().collect();
+                    let res = root_bank.rc.accounts.load_slow(&ancestors, pubkey);
+
+                    if let Some((mut account, _)) = res {
+                        if account.owner == solana_vote_program::id() {
+                            VoteStateVersions::convert_from_raw(&mut account, pubkey);
+                            root_bank.rc.accounts.store_slow(*slot, pubkey, &account);
+                            (*slot, pubkey, &account);
+                        }
+                    }
+                }
+            }
+
+            // Update the bank hash
+            println!("Updating bank hash");
+            root_bank.rc.accounts.accounts_db.update_accounts_hash(
```

Comment:
After an ugly merge, this seems to be working on my multinode test @sakridge, whoooo